calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
All sounds a bit weird...  Do you expect we'll rely on autodetection for the daemonsetted calico-node then?  We could specify it as an env option in kubernetes using a valueFrom:fieldRef:fieldPath:status.podIP instead (is there a hostname equivalent?)?

In a way I don't like calico-node doing any detection itself: I'd prefer it was pushed back to parameters (e.g. -e) rather than relying on side effects from the container inspecting the host networking.